File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/features/customers/createCustomer/CreateCustomerRequest.java
package br.com.deltaglobalbank.customers.features.customers.createCustomer;

import java.time.LocalDate;
import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CreateCustomerRequest(
    @NotBlank String cpf,
    @NotBlank String fullName,
    @NotNull LocalDate birthDate,
    @NotBlank String gender,
    String nationality,
    @NotBlank String motherName,
    @NotBlank String maritalStatus,
    String email,
    @NotNull @Valid PhoneRequest phone,
    @NotNull @Valid AddressRequest address,
    @Valid List<DocumentRequest> documents,
    @Valid List<BankAccountRequest> bankAccounts
) {
    public CreateCustomerRequest {
        if (nationality == null) {
            nationality = "brasileira";
        }
        if (documents == null) {
            documents = List.of();
        }
        if (bankAccounts == null) {
            bankAccounts = List.of();
        }
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/features/customers/createCustomer/CreateCustomerRequest.kt
package br.com.deltaglobalbank.customers.features.customers.createCustomer

import java.time.LocalDate

data class CreateCustomerRequest(
    val cpf: String,
    val fullName: String,
    val birthDate: LocalDate,
    val gender: String,
    val nationality: String = "brasileira",
    val motherName: String,
    val maritalStatus: String,
    val email: String? = null,
    val phone: PhoneRequest,
    val address: AddressRequest,
    val documents: List<DocumentRequest> = emptyList(),
    val bankAccounts: List<BankAccountRequest> = emptyList(),
)

data class PhoneRequest(
    val phoneNumber: String,
)

data class AddressRequest(
    val cep: String,
    val street: String,
    val number: String? = null,
    val complement: String? = null,
    val neighborhood: String? = null,
    val city: String,
    val state: String,
    val country: String = "BR",
)

data class DocumentRequest(
    val type: String,
    val number: String,
    val issuer: String,
    val issuerState: String,
    val issuedAt: LocalDate,
)

data class BankAccountRequest(
    val bankCode: String,
    val agency: String,
    val accountNumber: String,
    val accountDigit: String? = null,
    val accountType: String,
    val purpose: String,
    val isPrimary: Boolean = false,
)